api/envoy/config/wasm/v2/wasm.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.config.wasm.v2;
+
+option java_outer_classname = "WasmProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.config.wasm.v2";
+
+import "envoy/api/v2/core/base.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Wasm service]
+
+// Configuration for a Wasm VM.
+// [#next-free-field: 6]
+message VmConfig {
+  // An ID which will be used along with a hash of the wasm code (or null_vm_id) to determine which
+  // VM will be used for the plugin. All plugins which use the same vm_id and code will use the same
+  // VM. May be left blank.
+  string vm_id = 1;
+
+  // The Wasm runtime type (see source/extensions/common/wasm/well_known_names.h).
+  string runtime = 2;
+
+  // The Wasm code that Envoy will execute.
```

Comment:
I think we need a WASM section in the Envoy architecture intro or configuration docs that we link to that explains things like file format etc.

Reply:
@PiotrSikora can we push some of the ABI/docs to the envoyproxy/envoy repo?